src/components/accession2/Accession2Create.tsx

```diff
@@ -67,28 +80,48 @@ export default function CreateAccession(props: CreateAccessionProps): JSX.Elemen
     ...marginTop,
   };
 
+  const setDateError = (id: string, error?: string) => {
+    setDateErrors((prev) => ({
+      ...prev,
+      [id]: error,
+    }));
+  };
+
   const changeDate = (id: string, value?: any) => {
     setDates((curr) => ({ ...curr, [id]: value }));
     const date = new Date(value).getTime();
     const now = Date.now();
-    if (isNaN(date) || date > now) {
+
+    setDateError(id, '');
+
+    if (isNaN(date)) {
+      setDateError(id, strings.INVALID_DATE);
+      return;
+    } else if (date > now) {
+      setDateError(id, strings.NO_FUTURE_DATES);
       return;
     } else {
       onChange(id, value);
     }
   };
 
-  const getAccessionStatuses = () => {
-    // TODO: return statuses that can be used for create, for now return all statuses
-    return ACCESSION_2_STATES;
-  };
-
   const goToAccessions = () => {
     history.push(accessionsDatabase);
   };
 
+  const hasErrors = () => {
+    const missingRequiredField = MANDATORY_FIELDS.some((field: MandatoryField) => record[field] === undefined);
+    const hasDateErrors = dateErrors.recordedDate || dateErrors.collectedDate;
+    return missingRequiredField || hasDateErrors;
+  };
+
   const saveAccession = async () => {
-    // TODO data validation and show errors
+    if (hasErrors()) {
+      setDateError('collectedDate', record.collectedDate ? dateErrors.collectedDate : strings.REQUIRED_FIELD);
```

Comment:
Because we are changing the record only when the value exists, it can happen that if date was not set (undefined/null) and the you set and invalid date (5555-55-55), you get "invalid date" but after pressing save, this line changes the "Invalid date" error to "required field". Not very serious

Reply:
Excellent! Thank you!